modules/auxiliary/gather/piwigo_auth_sqli.rb

```diff
@@ -0,0 +1,115 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+    include Msf::Exploit::Remote::HttpClient
+  
+    def initialize(info = {})
+      super(update_info(info,
+        'Name'           => 'Piwigo Gather Credentials via SQL Injection',
+        'Description'    => %q{
+           This module enables an authenticated user to collect the usernames and
+           encrypted passwords of other users in the Piwigo via SQL
+           injection.
+        },
+        'Author'         => [
+                              'Rodolfo "bod4k" Tavares'
+                            ],
+        'License'        => MSF_LICENSE,
+        'References'     => [
+                              [ 'CVE', '2023-26876' ] # CVE-2023-26876 https://nvd.nist.gov/vuln/detail/CVE-2023-26876
+                            ],
+        'DisclosureDate' => '04/21/2023'
+      ))
+  
+      register_options(
+        [
+          OptString.new('TARGETURI', [ true, 'The base path to Piwigo', '/' ]),
+          OptString.new('USERNAME', [ true, 'The username for authenticating to Piwigo', 'piwigo' ]),
+          OptString.new('PASSWORD', [ true, 'The password for authenticating to Piwigo', 'piwigo' ])
+        ])
+    end
+  
+    def check_availability
```

Comment:
```suggestion
    def check
```
Could we convert this into a check method and return checkcodes here?

Reply:
fixed!

Reply:
does this work or does it need `check_host`?

Reply:
it works. I tried without `check_host` method from [https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-write-a-check-method.html] and works.